From 290b15873d410b4e5e231b96f0407c862026a57f Mon Sep 17 00:00:00 2001
From: aspeedyh <yh_chung@aspeedtech.com>
Date: Tue, 3 Mar 2026 10:54:34 +0800
Subject: [PATCH] edaf_bridge: clear SAFS size setting for channel 3

default SAFS size is set to 32KB, set to zero to disable size check to
avoid potential issue when SAFS is enabled.

Signed-off-by: aspeedyh <yh_chung@aspeedtech.com>
Change-Id: I7b9fcda485260b6d77143fa08e8d1967f6eb385d
(cherry picked from commit b08a81a871ea1060e4036e85ab266257bbb2c86c)

Signed-off-by: santhoshs <santhoshs@ami.com>
---
 drivers/misc/aspeed_edaf_bridge.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/misc/aspeed_edaf_bridge.c b/drivers/misc/aspeed_edaf_bridge.c
index 061856da2f..31b8a3f6bd 100644
--- a/drivers/misc/aspeed_edaf_bridge.c
+++ b/drivers/misc/aspeed_edaf_bridge.c
@@ -25,6 +25,9 @@
 #define ESPI_CHN3_CTL			0x400
 #define   ESPI_CHN3_SW_READY		BIT(5)
 
+#define ESPI_CHN3_FILTER_CTL		0x500
+#define   ESPI_CHN3_FILTER_SAFS_SIZE	GENMASK(24, 16)
+
 #define EDAF_BDGE_CFG			0x0
 #define   EDAF_BDGE_CFG_CMD_EN		BIT(0)
 #define EDAF_BDGE_CBASE			0x20
@@ -246,6 +249,10 @@ static int aspeed_edaf_bridge_probe(struct udevice *dev)
 		espi = readl(espi_regs + ESPI_CHN3_CTL);
 		espi |= ESPI_CHN3_SW_READY;
 		writel(espi, espi_regs + ESPI_CHN3_CTL);
+
+		espi = readl(espi_regs + ESPI_CHN3_FILTER_CTL);
+		espi &= ~ESPI_CHN3_FILTER_SAFS_SIZE;
+		writel(espi, espi_regs + ESPI_CHN3_FILTER_CTL);
 	}
 
 	return 0;
-- 
2.34.1

